internal/service/cloudwatch/contributor_managed_insight_rule.go

```diff
@@ -0,0 +1,400 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	fwflex "github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	tfslices "github.com/hashicorp/terraform-provider-aws/internal/slices"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource("aws_cloudwatch_contributor_managed_insight_rule", name="Contributor Managed Insight Rule")
+// @Tags(identifierAttribute="arn")
+// @Testing(tagsTest=false)
+func newResourceContributorManagedInsightRule(_ context.Context) (resource.ResourceWithConfigure, error) {
+	r := &resourceContributorManagedInsightRule{}
+
+	return r, nil
+}
+
+const (
+	ResNameContributorManagedInsightRule = "Contributor Managed Insight Rule"
+)
+
+type resourceContributorManagedInsightRule struct {
+	framework.ResourceWithConfigure
+	framework.WithNoOpUpdate[resourceContributorManagedInsightRuleData]
+}
+
+func (r *resourceContributorManagedInsightRule) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_cloudwatch_contributor_managed_insight_rule"
+}
+
+func (r *resourceContributorManagedInsightRule) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			names.AttrARN: framework.ARNAttributeComputedOnly(),
+			names.AttrResourceARN: schema.StringAttribute{
+				Required: true,
+			},
+			"template_name": schema.StringAttribute{
+				Required: true,
+			},
+			names.AttrState: schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString("ENABLED"),
+			},
+			"rule_name": schema.StringAttribute{
+				Computed: true,
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+		},
+	}
+}
+
+func (r *resourceContributorManagedInsightRule) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().CloudWatchClient(ctx)
+
+	var plan resourceContributorManagedInsightRuleData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := &cloudwatch.PutManagedInsightRulesInput{
+		ManagedRules: []awstypes.ManagedRule{
+			{
+				ResourceARN:  plan.ResourceArn.ValueStringPointer(),
+				TemplateName: plan.TemplateName.ValueStringPointer(),
+			},
+		},
+	}
+	resp.Diagnostics.Append(fwflex.Expand(ctx, plan, in)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in.ManagedRules[0].Tags = getTagsIn(ctx)
+
+	if plan.State.ValueString() == "ENABLED" || plan.State.IsNull() {
```

Comment:
Might also be worth making `ENABLED` and `DISABLED` constants since they are referenced often